### ISSUE TITLE
CAT prepare --existing: Process gets killed during loading of files

### ISSUE THREAD

**Issue by ldj248** [dutilh/CAT#50]
I would like to create the necessary files to run CAT based on an existing diamond database (so only the fastaid2LCAtaxid and taxids_with_multiple_offspring files need to be created, as all other files are "found"). While running `CAT prepare --existing -d . -t .` the process gets killed during loading of the nr.gz file or the prot.accession2taxid file. I tried several times, but the killing happens at different time points. The loading of the nr.gz files takes more than an hour (sometimes the process is killed before it finishes) and the loading of the prot file gets killed after roughly an hour too (I never got past this point). Is it normal for this step to take such a long time? I assumed it would go rather quickly as the nr.dmnd file has already been created...  Does it require a lot of memory?

**Comment by ldj248** [comment]
I tried another time, now it worked :-) It took six hours with existing to generate those two files only. Should I worry about corrected headers or headers that do not have a taxid assigned? Will the later sequences be ignored or assigned to root?

**Comment by bastiaanvonmeijenfeldt** [comment]
Hi @ldj248,

I'm glad it worked!

You shouldn't worry about corrected headers, it's a correction by CAT for the fact that the first prot accession in the mapping file is not always the LCA, so the corrections indicate an improvement to plain nr!

What fraction of sequences had no taxid assigned? With the latest version of CAT it should be about 1-2%. We try to get it to 0% but nr and the mapping file are not completely in sync yet. Currently hits that have no taxid assigned are ignored by CAT, so they also don't count for the taxonomic algorithm.

Hope this is clear! Let me know if you have further questions.

Bastiaan

**Comment by ldj248** [comment]
Hi Bastiaan 

I got 0.0% without taxid, but corresponding to more than 100,000 headers, which seemed considerable, but obviously not compared to 1-2%. I guess we don't have to worry too much if they are not taken into account.  

Thank you for your quick reply!

**Comment by bastiaanvonmeijenfeldt** [comment]
OK, sounds good! The reason those proteins do not have taxids is either because they are from an NCBI database that is not included in prot.accession2taxid, or because they have just been added to nr and are not updated in prot.accession2taxid yet. The first option _should_ not happen anymore from v5.2 onwards; I've had extensive contact with NCBI about this, and they've managed to parse all their databases in one go. It used to be for example that any protein that did not have (depricated) gi taxids was excluded from prot.accession2taxid.

Long story short, 0.0% is nice! :)

**Comment by ldj248** [comment]
Okay great! Thanks for the clarification :-)
